### DISCUSSION TITLE
Executing Public Actions from a WalletClient

### DISCUSSION THREAD

**Discussion by devsquaaa** [wevm/viem#439] [Idea / Feature Request] [answered]
Title says it all.
It would be pretty useful to be able to execute public actions via a `WalletClient`. Unless I'm mistaken, I currently have to keep both a `PublicClient` and a `WalletClient`, which seems unnecessary (I wish I could just keep a single `WalletClient`).

**Comment by jxom** [comment, marked as answer]
The reason why we designed it to be separate is because wallet providers (Injected `window.ethereum`, WalletConnect v2, etc) may not provide a large majority of "node"/"public" RPC methods like `eth_call`, `eth_newFilter`, `eth_getLogs`, etc... So it is separate to guard the consumer from invoking the wrong methods.

**Reply by devsquaaa** [reply]
Makes sense, thanks!

**Reply by tmm** [reply]
Added to the [FAQ](https://github.com/wagmi-dev/viem/blob/main/.github/FAQ.md#why-doesnt-wallet-client-support-public-actions)
